src/main/scala/com/cognite/sdk/scala/common/login.scala

```diff
@@ -12,14 +13,39 @@ class Login[F[_]](val requestSession: RequestSession[F]) {
   implicit val loginStatusDecoder = deriveDecoder[LoginStatus]
   implicit val dataLoginStatusDecoder = deriveDecoder[DataLoginStatus]
   def status(): F[LoginStatus] =
-    requestSession
-      .sendCdf { request =>
-        request
-          .get(uri"${requestSession.baseUri}/login/status")
-          .response(asJson[DataLoginStatus])
-          .mapResponse {
-            case Left(value) => throw value.error
-            case Right(value) => value.data
-          }
-      }
+    new UrlValidator().isValid(requestSession.baseUri.toString()) match {
+      case true =>
+        try {
+          requestSession
+            .sendCdf { request =>
+              request
+                .get(uri"${requestSession.baseUri}/login/status")
+                .response(asJson[DataLoginStatus])
+                .mapResponse {
+                  case Left(value) => throw value.error
+                  case Right(value) => value.data
+                }
+            }
+        } catch {
+          case _: Throwable =>
+            throw CdpApiException(
```

Comment:
This isn't a CdpApiException, I think we should just throw a RuntimeException.

Reply:
Unless we can come up with something more appropriate based on what the underlying exception is.

Reply:
Currently it is catching unknown host, connection timeout, and the circe io parsing errors. I was trying to make it general because the circe io parsing error doesn't give any sort of helpful message. I could try to separate it into catching unknown hosts and timeouts, which give their own helpful error messages, and the default case is the parsing error that I can make a helpful message for